parser-typechecker/src/Unison/Codebase.hs

```diff
@@ -143,6 +153,26 @@ import qualified Unison.Util.Relation as Rel
 import Unison.Var (Var)
 import qualified Unison.WatchKind as WK
 
+-- | Get the shallow representation of the root branches without loading the children or
+-- history.
+getShallowRootBranch :: Monad m => Codebase m v a -> m (Either GetRootBranchError (V2.CausalBranch m))
+getShallowRootBranch codebase = do
+  hash <- getRootBranchHash codebase
+  getShallowBranchForHash codebase hash >>= \case
+    Nothing -> pure $ Left (CouldntLoadRootBranch . Cv.causalHash2to1 $ hash)
+    Just b -> pure (Right b)
+
+-- | Recursively descend into shallow branches following the given path.
+shallowBranchAtPath :: Monad m => Path -> V2Branch.CausalBranch m -> m (Maybe (V2Branch.CausalBranch m))
```

Comment:
This should probably be moved to `U.Codebase.Branch`, but doing so requires also moving `Unison.Codebase.Path` into `unison-core1`, so I'll defer that to another PR.